odl/trafos/convolution.py

```diff
@@ -0,0 +1,849 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Operators based on (integral) transformations."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+import numpy as np
+import scipy.signal as signal
+
+from odl.discr import (
+    DiscreteLp, uniform_discr, uniform_discr_fromdiscr)
+from odl.operator import Operator, OpDomainError, OpRangeError
+from odl.set import ComplexNumbers
+from odl.trafos import FourierTransform, DiscreteFourierTransform
+from odl.util.normalize import normalized_scalar_param_list, safe_int_conv
+
+
+__all__ = ('FourierSpaceConvolution', 'RealSpaceConvolution')
+
+
+_REAL_CONV_SUPPORTED_IMPL = ('scipy_convolve',)
+# TODO: get from fourier.py
+_FOURIER_CONV_SUPPORTED_IMPL = ('numpy', 'pyfftw')
+
+
+class ConvolutionBase(Operator):
```

Comment:
I thought in the beginning it would be more useful, but it turns out to be rather pointless. I'll remove it.